### DISCUSSION TITLE
BG Music Character State

### DISCUSSION THREAD

**Discussion by shinra358** [ikemen-engine/Ikemen-GO#1912] [Feature Request] [answered]
### Describe the solution you'd like

example: bgm.char.state = [4000,4999]
              bgm.char.life = skibbidy.ogg ; if not looping from default, life, time, or round bgms
              bgm.char.round = skibbidy.ogg 
              bgm.char.default = skibbidy.ogg
              bgm.char.time = skibbidy.ogg

              bgm.char.loopstart.life = 93855 ; if looping from default, life, time, or round bgms or if all bgms are in one file
              bgm.char.loopend.life = 9385578
              bgm.char.loopstart.round = 938556
              bgm.char.loopend.round = 93855787
              bgm.char.loopstart.default = 93855676
              bgm.char.loopend.default = 9385578787
              bgm.char.loopstart.time = 9385567676
              bgm.char.loopend.time = 938557878754

             bgm.char.fadein = 10 ; fadeout/fadein range
             bgm.char.fadeout = 10

detects the state of any character directly from the stage. If a char's state is within the range, music will play from a range indicated within the main stage music file or completely new file. When character is out of the range, the music blends back into the original stage theme. If one character enters the range and another enters after, music would not start of but just continue to play until no character is in the range anymore. This will have highest priority over the other stage musics. And there would be an option for each stage music feature so that it can be a variation of the low life, per round, and original bgms.

This is a feature in SF4 when doing Ultra combos.

**Comment by potsmugen** [comment, marked as answer]
Too specific I think. Can currently be done with modules however.
